packages/inter-protocol/test/auction/test-proportionalDist.js

```diff
@@ -0,0 +1,168 @@
+import { test as anyTest } from '@agoric/zoe/tools/prepare-test-env-ava.js';
+
+import '@agoric/zoe/exported.js';
+
+import { makeIssuerKit } from '@agoric/ertp';
+import { makeTracer } from '@agoric/internal';
+
+import { setUpZoeForTest, withAmountUtils } from '../supports.js';
+import { distributeProportionalShares } from '../../src/auction/auctioneer.js';
+
+/** @type {import('ava').TestFn<Awaited<ReturnType<makeTestContext>>>} */
+const test = anyTest;
+
+const trace = makeTracer('Test AuctContract', false);
+
+const makeTestContext = async () => {
+  const { zoe } = await setUpZoeForTest();
+
+  const currency = withAmountUtils(makeIssuerKit('Currency'));
+  const collateral = withAmountUtils(makeIssuerKit('Collateral'));
+
+  trace('makeContext');
+  return {
+    zoe: await zoe,
+    currency,
+    collateral,
+  };
+};
+
+test.before(async t => {
+  t.context = await makeTestContext();
+});
+
+const checkProportions = (
+  t,
+  amountsReturned,
+  rawDeposits,
+  rawExpected,
+  kwd = 'ATOM',
+) => {
+  const { collateral, currency } = t.context;
+
+  const rawExp = rawExpected[0];
+  t.is(rawDeposits.length, rawExp.length);
+
+  const [collateralReturned, currencyReturned] = amountsReturned;
+  const fakeCollateralSeat = harden({});
+  const fakeCurrencySeat = harden({});
+  const fakeReserveSeat = harden({});
+
+  debugger;
+  const deposits = [];
+  const expectedXfer = [];
+  for (let i = 0; i < rawDeposits.length; i += 1) {
+    const seat = harden({});
+    deposits.push({ seat, amount: collateral.make(rawDeposits[i]) });
+    const currencyRecord = { Currency: currency.make(rawExp[i][1]) };
+    expectedXfer.push([fakeCurrencySeat, seat, currencyRecord]);
+    const collateralRecord = { Collateral: collateral.make(rawExp[i][0]) };
+    expectedXfer.push([fakeCollateralSeat, seat, collateralRecord]);
+  }
+  const expectedLeftovers = rawExpected[1];
+  const leftoverCurrency = { Currency: currency.make(expectedLeftovers[1]) };
+  expectedXfer.push([fakeCurrencySeat, fakeReserveSeat, leftoverCurrency]);
+  expectedXfer.push([
+    fakeCollateralSeat,
+    fakeReserveSeat,
+    { Collateral: collateral.make(expectedLeftovers[0]) },
+    { [kwd]: collateral.make(expectedLeftovers[0]) },
+  ]);
+
+  const transfers = distributeProportionalShares(
+    collateral.make(collateralReturned),
+    currency.make(currencyReturned),
+    // @ts-expect-error mocks for test
+    deposits,
+    fakeCollateralSeat,
+    fakeCurrencySeat,
+    'ATOM',
+    fakeReserveSeat,
+    collateral.brand,
+  );
+
+  t.deepEqual(transfers, expectedXfer);
+};
+
+test('distributeProportionalShares', t => {
```

Comment:
making `checkProportions` a macro would fit well for these tests https://github.com/avajs/ava/blob/main/docs/01-writing-tests.md#reusing-test-logic-through-macros

note that the ses-ava breaks the `test.macro` helper so just use the older style

Reply:
done